src/base/bittorrent/dbresumedatastorage.cpp

```diff
@@ -780,3 +786,36 @@ void BitTorrent::DBResumeDataStorage::Worker::storeQueue(const QVector<TorrentID
             .arg(err.message()), Log::CRITICAL);
     }
 }
+
+void BitTorrent::DBResumeDataStorage::Worker::makeBackup() const
+{
+    try
+    {
+        const Path bakTmpPath = m_path + u".tmp"_qs;
+        Utils::Fs::removeFile(bakTmpPath);
+
+        auto db = QSqlDatabase::database(m_connectionName);
+        const QWriteLocker locker {&m_dbLock};
+
+        QSqlQuery query {db};
+        const QString pathPlaceholder = u":path"_qs;
+        const QString statement = u"VACUUM INTO %1;"_qs.arg(pathPlaceholder);
+        if (!query.prepare(statement))
+            throw RuntimeError(query.lastError().text());
+
+        query.bindValue(pathPlaceholder, bakTmpPath.toString());
+        if (!query.exec())
+            throw RuntimeError(query.lastError().text());
```

Comment:
Shouldn't you delete `bakTmpPath` file here too? The file is incomplete.

Reply:
> Shouldn't you delete `bakTmpPath` file here too? The file might be incomplete.

Do you think it's necessary? It will still be deleted before the next attempt.

Reply:
>Do you think it's necessary? 

Yes, some users are always depleting their free disk space like no tomorrow. This minuscule space might save them, who knows.

Reply:
Ok. Will take care of it in exception handler.